lib/Runtime/Library/JavascriptString.cpp

```diff
@@ -2191,36 +2191,70 @@ namespace Js
     {
         charcount_t count = pThis->GetLength();
 
+		const char16* inStr = pThis->GetString();
+		const char16* inStrLim = inStr + count;
+		const char16* i = inStr;
+
+        // Try to find out the chars that do not need casing (in the ASCII range)
+        if (toCase == ToUpper)
+        {
+            while (i < inStrLim)
+			{
+
+				// first range of ascii lower-case (97-122)
+				// second range of ascii lower-case (223-255)
+				// non-ascii chars (255+)
+				if ('a' <= *i) {
+					if (*i <= 'z') { break; }
+					if ('�' <= *i) { break; }
+				}
+
+                i++;
```

Comment:
@FremyCompany while you're waiting for us to run the extra tests and come back to this change after milestone 1.2, could you make a few style updates to your code?

In this block (especially above) there is some really wonky formatting. So that this change conforms to our coding style, can you change all tabs to spaces, make sure that all the indentation makes sense, and make sure your open braces are on lines by themselves.

(The places you have `if (...) { break; }` are fine as they are.)